x/epochs/types/hooks.go

```diff
@@ -42,13 +42,13 @@ func panicCatchingEpochHook(
 	epochIdentifier string,
 	epochNumber int64,
 ) {
-	defer func() {
-		if recovErr := recover(); recovErr != nil {
-			osmoutils.PrintPanicRecoveryError(ctx, recovErr)
-		}
-	}()
-	cacheCtx, write := ctx.CacheContext()
-	hookFn(cacheCtx, epochIdentifier, epochNumber)
-	write()
-	ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
+	wrappedHookFn := func(ctx sdk.Context) error {
```

Comment:
Can this function return err in any case?

Reply:
Not right now, it will after #2520

Reply:
Ah makes sense now

Reply:
have created draft #2526
waiting on this one, so it's cleaner